### PR TITLE
 [Optimize] change jwtParse to singleton with thread safety issues un…

### PR DESCRIPTION
**Pull request by dach-wh**
…it test

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/JwtTokenManager.java [hunk 1]
```diff
@@ -97,7 +101,7 @@ public Authentication getAuthentication(String token) {
      * @param token token
      */
     public void validateToken(String token) {
-        Jwts.parserBuilder().setSigningKey(nacosAuthConfig.getSecretKeyBytes()).build().parseClaimsJws(token);
+        parser.parseClaimsJws(token);
```
Comment (KomachiSion) [inline plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/JwtTokenManager.java:104]: Whether JwtParser is thread safety?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/JwtTokenManager.java (hunk 1 above)
Comment (dach-wh) [reply plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/JwtTokenManager.java:104]: I write a utit test to check the thread safety issues.
`    @Test
    public void testConcurrentCreateTokenAndSecretKeyWithoutSpecialSymbol() {
        //this unit test is to approve the singleton jwtParse don't have thread safety issues
        IntStream.range(0, 100).parallel().forEach(i -> {
            try {
                createToken(null);
            } catch (NoSuchFieldException | IllegalAccessException | RuntimeException e) {
                e.printStackTrace();
                //concurrent parse jwt failed, make unit test failed.
                Assert.isTrue(false);
            }
        });
    }`
 'singleton jwtParse' parse  jwttoken success in multithreading environment , so this unit test approve   'singleton jwtParse'  don`t have a thread safety issues

Comment (dach-wh) [reply plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/JwtTokenManager.java:104]: I debug the logic of JwtParser parse jwttoken,   JwtParser create a new SecretKeySpec object by clone the secretKeyBytes, so the  secretKey don`t have a thread safety issues.
image

**Comment by KomachiSion** [comment]
OK， I will merged if ci passed

**Comment by KomachiSion** [comment]
CI failed, please fix problem.

**Comment by dach-wh** [comment]
fixed code style

**Comment by KomachiSion** [comment]
IT can't pass for a long time and author has no more changes for this.

**Pull request closed without merging**